File: dynabuffers-python/dynabuffers/ast/annotation/GreaterThan.py
from dynabuffers.api.IAnnotation import IAnnotation
from dynabuffers.ast.structural import Value


class GreaterThan(IAnnotation):

    def __init__(self, args:[Value]):
        self.args = args
        if len(args) != 1:
            raise ValueError("too much annotation values")
        if not str(args[0].options.value).isnumeric():
            raise ValueError(str(args[0].options.value) + " is not a number")

    def validate(self, fieldName, value):
        if float(value) <= float(self.args[0].options.value):
            raise ValueError("(" + str(fieldName) + ":" + str(value) + ") is lower equals " + str(self.args[0].options.value))

File: dynabuffers-java/src/main/kotlin/dynabuffers/ast/annotation/GreaterThan.kt
package dynabuffers.ast.annotation

import dynabuffers.api.IAnnotation
import dynabuffers.ast.structural.Value
import dynabuffers.exception.DynabuffersException

class GreaterThan(private val args:List<Value>) : IAnnotation {

    init {
        require(args.size == 1)
        require(args[0].isNumeric()) {"${args[0].options.value} is not a number"}
    }

    override fun validate(fieldName:String, obj:Any?) {
        if (obj.toString().toFloat() <= args[0].asFloat()) {
            throw DynabuffersException("($fieldName:'$obj') is lower equals ${args[0].asFloat()}")
        }
    }
}
